src/components/patch-details/patch-details.tsx

```diff
@@ -17,16 +32,114 @@ interface PatchDetailsProps {
   assetPk: string;
   assetPatch?: Patch;
   assetArea?: Patch;
+  versionNumber?: number;
 }
 
-export const PatchDetails = ({ assetPk, assetPatch, assetArea }: PatchDetailsProps) => {
+export const buildUpdatePropertyPatchRequest = (
+  patchId: string,
+  areaId: string,
+): UpdatePropertyPatchRequest => ({
+  patchId,
+  areaId,
+});
+
+export const PatchDetails = ({
+  assetPk,
+  assetPatch,
+  assetArea,
+  versionNumber,
+}: PatchDetailsProps) => {
   const { heading } = locale.patchDetails;
 
   const { patchLabel, housingOfficerLabel, areaManagerLabel } = locale.patchDetails;
 
   const patchOrAreaDefined = assetPatch || assetArea;
   const housingOfficerName = assetPatch?.responsibleEntities[0]?.name;
   const areaManagerName = assetArea?.responsibleEntities[0]?.name;
+  const [newPatchName, setNewPatchName] = useState<string>();
+  const [isEditingPatchName, setIsEditingPatchName] = useState(false);
+  const [patchesAndAreas, setPatchesAndAreas] = useState<Patch[]>([]);
+
+  useEffect(() => {
+    getAllPatchesAndAreas().then((data) => {
+      setPatchesAndAreas(
+        data.filter(
+          (patchOrArea) =>
+            ![
+              "Hackney",
+              "CL Area",
+              "CP Area",
+              "HN1 Area",
+              "HN2 Area",
+              "SD Area",
+              "SH Area",
+              "SN Area",
+            ].includes(patchOrArea.name),
+        ),
+      );
+    });
+  }, []);
+
+  const updatePropertyPatchCall = (request: UpdatePropertyPatchRequest) => {
+    updatePropertyPatch(assetPk, request, versionNumber?.toString() ?? "")
+      .catch((err) => {
+        console.error(err);
+      })
+      .finally(() => {
+        window.location.reload();
+      });
+  };
+  const getbyPatchNameCall = (patchName: string) => {
+    getByPatchName(patchName)
+      .then((data) => {
+        const id = JSON.stringify(data).match(/"id":"(.*?)"/);
+        const parentId = JSON.stringify(data).match(/"parentId":"(.*?)"/);
+        console.log("id L84", id);
+        console.log("parentId L85", parentId);
+        if (!id?.[1] || !parentId?.[1]) {
+          throw new Error("Invalid patch data: id or parentId is undefined");
+        }
+        const request = buildUpdatePropertyPatchRequest(id[1], parentId[1]);
+        updatePropertyPatchCall(request);
+      })
+      .catch((err) => {
+        console.error(err);
```

Comment:
How critical is this action?
If this update failed, would we want the user to know?